conf/confc.c

```diff
@@ -481,6 +482,52 @@ static int confc_cache_create(struct m0_confc *confc,
 	return M0_RC(rc);
 }
 
+M0_INTERNAL int m0_confc_cache_add_service(struct m0_conf_cache    *cache,
+					   struct m0_conf_obj      *base_obj,
+					   struct m0_conf_obj      **new_obj,
+					   const struct m0_ha_note *nv_note)
+{
+	M0_LOG(M0_ERROR,"This is dummy function to add dynamic service fid"
+		        "Ideally we should not land here until dynamic service"
+		        "FID's are implemented");
+	return M0_RC(0);
+}
+
+M0_INTERNAL int m0_confc_cache_add_process(struct m0_conf_cache    *cache,
+					   struct m0_conf_obj      *base_obj,
+					   struct m0_conf_obj      **new_obj,
+					   const struct m0_ha_note *nv_note)
+{
+	struct m0_conf_dir     *dir;
+	struct m0_conf_process *proc1;
+	struct m0_conf_process *proc2;
+	int                     rc;
+
+	rc = m0_conf_obj_find(cache, &nv_note->no_id, new_obj);
+	if (rc == 0) {
+		(*new_obj)->co_ha_state = nv_note->no_state;
+
+		if ((*new_obj)->co_parent == NULL) {
+			dir = M0_CONF_CAST(base_obj->co_parent, m0_conf_dir);
+			m0_conf_dir_add(dir, *new_obj);
+		}
+		proc1 = M0_CONF_CAST(base_obj, m0_conf_process);
+		proc2 = M0_CONF_CAST(*new_obj, m0_conf_process);
+
+		if (proc1 != NULL && proc2 != NULL) {
```

Comment:
process = M0_CONF_CAST(obj, m0_conf_process);
        rc = m0_bitmap_init(&process->pc_cores, cores->b_nr);
        if (rc != 0)
                goto fail;
        /* XXX FIXME: process->pc_cores.b_words will leak in case of error */
        m0_bitmap_copy(&process->pc_cores, cores);
        process->pc_memlimit_as      = memlimit_as;
        process->pc_memlimit_rss     = memlimit_rss;
        process->pc_memlimit_stack   = memlimit_stack;
        process->pc_memlimit_memlock = memlimit_memlock;
        process->pc_endpoint         = m0_strdup(endpoint);
        if (process->pc_endpoint == NULL) {
                rc = M0_ERR(-ENOMEM);
                goto fail;
        }
Refer m0_spiel_process_add()

Reply:
referred m0_spiel_process_add() and updated patch to make it bit similar to this.